terraform/transform_attach_schema.go

```diff
@@ -60,14 +62,25 @@ func (t *AttachSchemaTransformer) Transform(g *Graph) error {
 			mode := addr.Resource.Mode
 			typeName := addr.Resource.Type
 			providerAddr, _ := tv.ProvidedBy()
-			var providerFqn addrs.Provider
 
+			var providerTypeName string
 			switch p := providerAddr.(type) {
 			case addrs.LocalProviderConfig:
-				// FIXME: need to look up the providerFQN in the config
-				providerFqn = addrs.NewLegacyProvider(p.LocalName)
+				providerTypeName = p.LocalName
 			case addrs.AbsProviderConfig:
-				providerFqn = p.Provider
+				providerTypeName = p.Provider.Type
```

Comment:
I'm not sure if I'm misreading the control flow here, but this feels a bit counter-intuitive... it looks like we're taking the type portion of an FQN here and then below possibly using it as a `LocalName` (on line 82).

I think what I was expecting here is that all of the logic below related to building `providerFqn` would be bypassed if `providerAddr` is `addrs.AbsProviderConfig`, because if we already have an absolute provider config address then `providerFqn` would be `p.Provider` here, right? :thinking:

Reply:
Yikes! You're absolutely right, that's a function that didn't get properly refactored after *I* refactored `ProvidedBy()`.  I've missed `AttachSchemaTransformer` too many times to count _already_ - thank you